### ISSUE TITLE
Postprocessing JS

### ISSUE THREAD

**Issue by jeswr** [emscripten-core/emscripten#18740]
The current JS that emscripten generates uses some old conventions + often generates unused code; I have seen a few issues going around offering manual updates of some of these things.

I wanted to note that a combination of tools like [`lebab`](https://github.com/lebab/lebab), `eslint` and `rollup` could be used in a post processing step to tidy up the `.js` files that emscripten emits.

In particular, with the default `eslint` and `rollup` configs I just ran on a generated file called `swipl.js`

```ts
lebab --replace swipl.js --transform arrow,arrow-return,for-of,arg-rest,arg-spread,obj-method,obj-shorthand,no-strict,exponent,multi-var,let,class,commonjs,template,default-param,destruct-param,includes
eslint swipl.js
rollup swipl.js -o out.js
```

The `out.js` was about 10% smaller on the project I tried and has various code smells removed such as limiting the use of `var`, removing unused variables etc.

**Comment by sbc100** [comment]
Interesting.  Were you able to improve code size even with `--closure=1` and `-O3`?   I think we generally worry more about code size than code smells in the generated JS, since the output JS can kind of be considered machine-generated code (of course it would be nice if looked nice too)

**Comment by jeswr** [comment]
Will double check my settings - the main code removed seemed to be unused variable declarations in the Javascript.

**Comment by sbc100** [comment]
Yeah I think we generally rely on closure compiler to remove all of those.     Or we try not to generate them in first place (by using our fake C pre-processor thing).
